components/gfx/text/text_run.rs

```diff
@@ -302,6 +324,14 @@ impl<'a> TextRun {
         })
     }
 
+    pub fn on_glyph_run_boundary(&self, index: ByteIndex) -> bool {
+        if let Some(glyph_index) = self.index_of_first_glyph_run_containing(index) {
+            self.glyphs[glyph_index].range.begin() == index
+        } else {
+            true
```

Comment:
I *think* this case should never be reached.  If that's correct, maybe it should panic.